app/models/vendor.rb

```diff
@@ -73,4 +73,17 @@ def status
       product_counts.key?(product_state) ? product_counts[product_state].count : 0
     end
   end
+
+  def header_fields?
+    url? || address? || !points_of_contact.empty?
+  end
+
+  def product_arr
+    Product.where(vendor_id: id).by_updated_at.order_by(state: 'desc')
+  end
+
+  def favorite_products(current_user)
```

Comment:
We have favorites for both products and vendors but they are both done very differently. Currently the vendors are all gathered in 1 array and then filtered in the view, and the products are selected from the DB twice. Is this just due to us only updating 1 of the methods and not the other?